src/globalvars.cpp

```diff
@@ -258,19 +320,143 @@ void GlobalVars::verify_maps_found(
                << " which is needed to set global variable " << name;
     }
   }
 }
 
+Result<GlobalVarMap> GlobalVars::get_named_param_vals(
+    std::vector<std::string> raw_named_params) const
+{
+  std::unordered_map<std::string, std::optional<std::string>> named_params;
+  std::vector<std::string> unexpected_params;
+
+  for (const auto &param : raw_named_params) {
+    if (param.find("=") != std::string::npos) {
+      auto split = util::split_string(param,
+                                      '=',
+                                      /* remove_empty= */ true);
+
+      named_params[split[0]] = split.size() == 2 ? split[1] : "";
+      if (!added_global_vars_.contains(split[0])) {
+        unexpected_params.push_back(split[0]);
+      }
+    } else {
+      named_params[param] = std::nullopt;
+      if (!added_global_vars_.contains(param)) {
+        unexpected_params.push_back(param);
+      }
+    }
+  }
+
+  if (!unexpected_params.empty()) {
+    std::vector<std::string> expected_params;
+    for (const auto &[name, _] : named_param_defaults_) {
+      expected_params.push_back(name);
+    }
+
+    return make_error<UnknownParamError>(std::move(unexpected_params),
+                                         std::move(expected_params));
+  }
+
+  GlobalVarMap named_param_vals;
+
+  for (const auto &[name, default_val] : named_param_defaults_) {
+    const auto &config = added_global_vars_.at(name);
+    auto it = named_params.find(name);
+
+    // Use the default
+    if (it == named_params.end()) {
+      if (config.type == Type::integer) {
+        named_param_vals[name] = std::get<int64_t>(default_val);
+      } else if (config.type == Type::boolean) {
+        named_param_vals[name] = std::get<bool>(default_val);
+      } else if (config.type == Type::string) {
+        named_param_vals[name] = std::get<std::string>(default_val);
+      }
+      continue;
+    }
+
+    // No value means it's a boolean
+    if (!it->second.has_value()) {
+      if (config.type == Type::integer) {
+        return make_error<NamedParamError>(name, "true", "expects an integer");
+      } else if (config.type == Type::string) {
+        return make_error<NamedParamError>(name, "true", "expects a string");
+      } else if (config.type == Type::boolean) {
+        named_param_vals[name] = true;
+      }
+      continue;
+    }
+
+    const std::string &val = *it->second;
+
+    if (config.type == Type::integer) {
+      if (val[0] == '-') {
+        auto int_val = util::to_uint(val.substr(1, val.size() - 1));
+        if (!int_val) {
+          return make_error<NamedParamError>(name, val, "expects an integer");
+        }
+        // Converting without overflow.
+        auto neg = -static_cast<int64_t>(*int_val - 1) - 1;
+        if (neg > 0) {
+          return make_error<NamedParamError>(name,
+                                             val,
+                                             "value is out of range");
+        }
```

Comment:
This should be IMHO moved to `util::to_int`.

Reply:
Yeah, I agree. Let me see about refactoring.

Reply:
Pulled it into the int_parser util